activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -624,6 +626,26 @@ def check_current_protected_environment!(db_config)
           rescue ActiveRecord::NoDatabaseError
           end
         end
+
+        def with_env(env)
```

Comment:
I'm going to take a look at this this week. We can't merge this method as is, we need to do this without accessing Rails env at all.